### PR TITLE
Risky food simulation

### PR DESCRIPTION
**Pull request by rlskoeser**
implements the "risky food" game as described in #3 

- includes preliminary readme describing the model and how to run it
- preliminary data collection for reporting/analysis
- prelininary unit testing to confirm logic

### CODE REVIEW

File: tests/test_risky_food.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        results.append(model.get_risky_food_status())
+
+    # use counter to tally the results
+    result_count = Counter(results)
+
+    # the expected value is the probability times number of times we ran it
+    expected = total_runs * model.prob_notcontaminated
+    assert math.isclose(
+        result_count[FoodStatus.NOTCONTAMINATED], expected, abs_tol=total_runs * 0.1
+    )
```
Comment (quadrismegistus) [inline tests/test_risky_food.py:36]: This is nice. 'Testing' takes on an interesting meaning here. One knows (from the scholarship) what expected outcomes of particular models ought to be. These can be written into our implementations directly as tests.

File: simulatingrisk/risky_food/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- Nature selects a probability `p` for **N**
+- For each agent: if `r` > `p`, then they choose RISKY; else SAFE
+- Nature flips a coin with bias `p` for **N**, and announces **N** or **C**
+- If **N**: everyone who chose RISKY gets 3, everyone who chose SAFE gets 2
+- If **C**: everyone who chose RISKY gets 1, everyone SAFE 2
+- Reproduce in proportion to payoff
+  - Either agent gets # of offspring = payoff [they replace–original “dies off”]
+  - OR: take the total payoff for RISKYs over total for everyone, there are that proportion of RISKYs in the new population
+
+END ROUND
```
Comment (quadrismegistus) [inline simulatingrisk/risky_food/README.md:22]: I like this semi-formal way of describing a 'round'! Very clear articulation of the steps/rules of the game.

File: simulatingrisk/risky_food/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+SEE: We’ll see what are the risk attitudes that are replicated more and less over time
+
+## Running the simulation
+
+- Install python dependencies as described in the main project readme (requires mesa)
+- To run from the main `simulating-risk` project directory: 
+	- Configure python to include the current directory in import path; 
+	  for C-based shells, run `setenv PYTHONPATH .` ; for bash, run `export $PYTHONPATH=.`
+	- To run interactively with mesa runserver: `mesa runserver simulatingrisk/risky_food/`
```
Comment (quadrismegistus) [inline simulatingrisk/risky_food/README.md:32]: Works for me!

File: simulatingrisk/risky_food/model.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        # get a random risk tolerance; returns a value between 0.0 and 1.0
+        self.risk_level = risk_level or self.random.random()
+
+    def step(self):
+        # choose food based on the probability not contaminated and risk tolerance
+        if self.risk_level > self.model.prob_notcontaminated:
+            choice = FoodChoice.RISKY
+        else:
+            choice = FoodChoice.SAFE
+        self.payoff = self.model.payoff(choice)
```
Comment (quadrismegistus) [inline simulatingrisk/risky_food/model.py:29]: Again I feel like a 1st-person POV might be more legible as a way of describing what's happening

File: simulatingrisk/risky_food/model.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            model_reporters={
+                "prob_notcontaminated": "prob_notcontaminated",
+                "contaminated": "contaminated",
+                "average_risk_level": "avg_risk_level",
+                "min_risk_level": "min_risk_level",
+                "max_risk_level": "max_risk_level",
+                "num_agents": "total_agents",
+            },
+            agent_reporters={"risk_level": "risk_level", "payoff": "payoff"},
+        )
```
Comment (quadrismegistus) [inline simulatingrisk/risky_food/model.py:55]: Interesting setup they provid

**Review by quadrismegistus — APPROVED** [review]
Really exciting work. Runs perfectly and code reads well!

File: simulatingrisk/risky_food/model.py (hunk 5 above)
Comment (rlskoeser) [reply simulatingrisk/risky_food/model.py:55]: seems very verbose / redundant to me, but at least workable enough for now

File: simulatingrisk/risky_food/model.py (hunk 4 above)
Comment (rlskoeser) [reply simulatingrisk/risky_food/model.py:29]: can you say more, I don't understand?

File: simulatingrisk/risky_food/README.md (hunk 2 above)
Comment (rlskoeser) [reply simulatingrisk/risky_food/README.md:22]: this is all from Lara! I just added some formatting. very clear, right?

File: simulatingrisk/risky_food/model.py (hunk 4 above)
Comment (rlskoeser) [reply simulatingrisk/risky_food/model.py:29]: oh, read in the wrong order - I see your comments on the other PR now

**Pull request merged**